optimade/server/entry_collections/entry_collections.py

```diff
@@ -43,22 +117,120 @@ def get_attribute_fields(self) -> set:
+        if getattr(params, "filter", False):
+            tree = self.parser.parse(params.filter)
+            cursor_kwargs["filter"] = self.transformer.transform(tree)
+        else:
+            cursor_kwargs["filter"] = {}
 
-        Args:
-            params (EntryListingQueryParams): entry listing URL query params
+        if (
+            getattr(params, "response_format", False)
+            and params.response_format != "json"
+        ):
+            raise BadRequest(
+                detail=f"Response format {params.response_format} is not supported, please use response_format='json'"
+            )
+
+        if getattr(params, "page_limit", False):
+            limit = params.page_limit
+            if limit > CONFIG.page_limit_max:
+                raise Forbidden(
+                    detail=f"Max allowed page_limit is {CONFIG.page_limit_max}, you requested {limit}",
+                )
+            cursor_kwargs["limit"] = limit
+        else:
+            cursor_kwargs["limit"] = CONFIG.page_limit
+
+        cursor_kwargs["fields"] = self.all_fields
+        cursor_kwargs["projection"] = [
+            self.resource_mapper.alias_for(f) for f in self.all_fields
+        ]
+
+        if getattr(params, "sort", False):
+            cursor_kwargs["sort"] = self.parse_sort_params(params.sort)
+
+        if getattr(params, "page_offset", False):
+            cursor_kwargs["skip"] = params.page_offset
+
+        return cursor_kwargs
+
+    def parse_sort_params(self, sort_params) -> List[Tuple[str, int]]:
+        """ Handles any sort parameters passed to the collection,
+        resolving aliases and dealing with any invalid fields.
+
+        Raises:
+            BadRequest: if an invalid sort is requested.
 
         Returns:
-            Tuple[List[Entry], int, bool, set]: (results, data_returned, more_data_available, fields)
+            A list of tuples containing the aliased field name and
+            sort direction encoded as 1 (ascending) or -1 (descending).
 
         """
+        sort_spec = []
+        for field in sort_params.split(","):
+            sort_dir = 1
+            if field.startswith("-"):
+                field = field[1:]
+                sort_dir = -1
+            aliased_field = self.resource_mapper.alias_for(field)
+            sort_spec.append((aliased_field, sort_dir))
+
+        unknown_fields = [
+            field
+            for field, _ in sort_spec
+            if self.resource_mapper.alias_of(field) not in self.all_fields
+        ]
+
+        if unknown_fields:
+            error_detail = "Unable to sort on unknown field{} '{}'".format(
+                "s" if len(unknown_fields) > 1 else "", "', '".join(unknown_fields),
+            )
+
+            # If all unknown fields are "other" provider-specific, then only provide a warning
+            if all(
+                (
+                    field.startswith("_")
```

Comment:
The `is not None` part should theoretically be removable, buuuut 😅 
Also, maybe the last `+` could strictly be a `*` instead?
```suggestion
                    re.match(r"_[a-z]+_[a-z_0-9]+", field) is not None
```

Reply:
Fair point, the provider prefix can be alphanumeric too though right? Think I'm reading the spec correctly there